Tools/jit/build.py

```diff
@@ -0,0 +1,695 @@
+"""A template JIT for CPython 3.13, based on copy-and-patch."""
+
+# XXX: I'll probably refactor this file a bit before merging to make it easier
+# to understand... it's changed a lot over the last few months, and the way
+# things are done here definitely has some historical cruft. The high-level
+# architecture and the actual generated code probably won't change, though.
+
+import argparse
+import asyncio
+import dataclasses
+import enum
+import hashlib
+import json
+import os
+import pathlib
+import re
+import shlex
+import subprocess
+import sys
+import tempfile
+import typing
+
+import llvm
+import schema
+
+if sys.version_info < (3, 11):
+    raise RuntimeError("Building the JIT compiler requires Python 3.11 or newer!")
+
+TOOLS_JIT_BUILD = pathlib.Path(__file__).resolve()
+TOOLS_JIT = TOOLS_JIT_BUILD.parent
+TOOLS = TOOLS_JIT.parent
+CPYTHON = TOOLS.parent
+INCLUDE = CPYTHON / "Include"
+INCLUDE_INTERNAL = INCLUDE / "internal"
+PYTHON = CPYTHON / "Python"
+PYTHON_EXECUTOR_CASES_C_H = PYTHON / "executor_cases.c.h"
+TOOLS_JIT_TEMPLATE_C = TOOLS_JIT / "template.c"
+
+
+@enum.unique
+class HoleValue(enum.Enum):
+    CODE = enum.auto()
+    CONTINUE = enum.auto()
+    DATA = enum.auto()
+    EXECUTOR = enum.auto()
+    GOT = enum.auto()
+    OPARG = enum.auto()
+    OPERAND = enum.auto()
+    TARGET = enum.auto()
+    TOP = enum.auto()
+    ZERO = enum.auto()
+
+
+@dataclasses.dataclass
+class Hole:
+    offset: int
+    kind: schema.HoleKind
+    value: HoleValue
+    symbol: str | None
+    addend: int
+    replace = dataclasses.replace
+
+
+S = typing.TypeVar("S", schema.COFFSection, schema.ELFSection, schema.MachOSection)
+R = typing.TypeVar(
+    "R", schema.COFFRelocation, schema.ELFRelocation, schema.MachORelocation
+)
+
+
+@dataclasses.dataclass
+class Stencil:
+    body: bytearray = dataclasses.field(default_factory=bytearray)
+    holes: list[Hole] = dataclasses.field(default_factory=list)
+    disassembly: list[str] = dataclasses.field(default_factory=list)
+    symbols: dict[str, int] = dataclasses.field(default_factory=dict, init=False)
+    offsets: dict[int, int] = dataclasses.field(default_factory=dict, init=False)
+    relocations: list[Hole] = dataclasses.field(default_factory=list, init=False)
+
+    def pad(self, alignment: int) -> None:
+        offset = len(self.body)
+        padding = -offset % alignment
+        self.disassembly.append(f"{offset:x}: {' '.join(['00'] * padding)}")
+        self.body.extend([0] * padding)
+
+    def emit_aarch64_trampoline(self, hole: Hole) -> typing.Generator[Hole, None, None]:
```

Comment:
```suggestion
    def emit_aarch64_trampoline(self, hole: Hole) -> typing.Iterable[Hole]:
```

Most likely that is the only thing you care about in this contract.